scripts/validation/embedded-examples-validation.js

```diff
@@ -0,0 +1,119 @@
+function extractCommentsAndExamples(content) {
+  const combinedRegex = /<!--\s*asyncapi-example-tester:\s*({.*?})\s*-->\s*\n```(.*)?\n([\s\S]*?)\n```/g;
+  let match;
+  const combinedData = [];
+
+  while ((match = combinedRegex.exec(content)) !== null) {
+    try {
+      const json = JSON.parse(match[1]);
+      const format = match[2].trim();
+      const exampleContent = match[3].trim();
+
+      let example;
+      if (format === 'json') {
+        example = JSON.parse(exampleContent);
+      } else if (format === 'yaml') {
+        example = yaml.load(exampleContent);
+      } else {
+        throw new Error(`Unsupported format: ${format}`);
+      }
+
+      combinedData.push({
+        name: json.name,
+        json_path: json.json_path,
+        example: example,
+        format: format,
+      });
+    } catch (e) {
+      console.error("Failed to parse comment JSON or example:", match[1], e);
+      // process.exit(1);
+    }
+  }
+
+  return combinedData;
+}
+
+// Extract comments and examples from the markdown file
+const combinedData = extractCommentsAndExamples(markdownContent);
+
+// Function to apply JSON Merge Patch updates to the document
+function applyUpdates(updates, baseDoc) {
+  updates.forEach(update => {
+    try {
+      if (update.json_path === "$") {
+        // Apply patch directly at the root
+        baseDoc = mergePatch.apply(baseDoc, update.example);
+      } else {
+        // Apply patch at a specified JSON Pointer path
+        const targetObject = jsonpointer.get(baseDoc, update.json_path);
+        const patchedObject = mergePatch.apply(targetObject || {}, update.example);
+        jsonpointer.set(baseDoc, update.json_path, patchedObject);
+      }
+    } catch (e) {
+      console.error(`\nError processing update for '${update.name}' at path '${update.json_path}'`, e);
+      // process.exit(1);
+    }
+  });
+  return baseDoc;
+}
+
+// Function to validate a document using AsyncAPI parser
+async function validateParser(document, name) {
+  try {
+    const diagnostics = await parser.validate(document);
+
+    if (diagnostics.length > 0) {
+      diagnostics.forEach(diagnostic => {
+        if (diagnostic.level === 'error') {
+          console.error(`\x1b[31mError in ${name}: ${diagnostic.message}\x1b[0m`);
+        } else {
+          console.log(`\x1b[31mWarning in ${name}: ${diagnostic.message}\x1b[0m`);
+        }
+      });
+    } else {
+      console.log(`${name} is valid.`);
+    }
+  } catch (error) {
+    console.error(`\x1b[31mValidation failed for ${name}: ${error.message}\x1b[0m`);
+  }
+}
+
+// Iterate over the combinedData array, apply updates, and validate each document
+const baseDocPath = './base-doc.json';
+const baseDocSecuritySchemePath = './base-doc-security-scheme-object.json';
+
+const baseDoc = JSON.parse(fs.readFileSync(baseDocPath, 'utf8'));
+const baseDocSecurityScheme = JSON.parse(fs.readFileSync(baseDocSecuritySchemePath, 'utf8'));
```

Comment:
Can't be this merged into the general basedoc?

Reply:
the examples under the `security scheme object` as far as I've seen if they contain these objects, they fail the validation:
```json
"flows": {
  "clientCredentials": {
    "tokenUrl": "https://example.com/api/oauth/dialog",
    "availableScopes": {
      "subscribe:auth_revocations": "Scope required for authorization revocation topic"
    }
  }
},
"scopes": [
  "subscribe:auth_revocations"
]
```

So I though to separate the whole `security scheme object` examples and provide them with a separate doc with minimal objects to add upon, and by doing that, they are exactly constructed as expected and are, in the end, valid documents.

Reply:
Not ideal at all to maintain two base documents and have logic dedicated for those. We should work on making it work as just one doc. Happy to assist.

Reply:
Okay, let me try it for myself once. I'll let you know the results. :+1:

Reply:
Hi @smoya :wave: 
- Tried adding more cases to [the security scheme field](https://gist.github.com/AnimeshKumar923/45e1e38934964f8a64e4fe36dd061246#file-securityscheme-json). 
- Passing the validation as expected, but unsure about the structure of the `securityScheme`. Can you please verify it once? :sweat_smile: 
- Full base doc also given in the gist.

Reply:
@smoya gentle reminder

Reply:
> Maybe [this](https://github.com/AnimeshKumar923/asyncapi-spec/blob/54fab4bd1f172a81e43ecd6a2058a4160f1f4e06/spec/asyncapi.md?plain=1#L2614C1-L2634C52) is the potential fix to the **OAuth Flow Object**? 👀

Yes, it is. In fact, I suggest you to add a new subtitle here clarifying with kind of flow the example is representing. 
I would add 

```

###### clientCredentials Oauth Flow object example
```

Right after the `##### OAuth Flow Object Examples` at https://github.com/AnimeshKumar923/asyncapi-spec/blob/54fab4bd1f172a81e43ecd6a2058a4160f1f4e06/spec/asyncapi.md?plain=1#L2612

Reply:
Any update on [this](https://github.com/asyncapi/spec/pull/1059#discussion_r1648575632)? Having two files is completely not ideal.

Reply:
> Any update on [this](https://github.com/asyncapi/spec/pull/1059#discussion_r1648575632)? Having two files is completely not ideal.

@smoya 
About that, please refer to [this comment](https://github.com/asyncapi/spec/pull/1059#discussion_r1655149132)

Reply:
> > Maybe [this](https://github.com/AnimeshKumar923/asyncapi-spec/blob/54fab4bd1f172a81e43ecd6a2058a4160f1f4e06/spec/asyncapi.md?plain=1#L2614C1-L2634C52) is the potential fix to the **OAuth Flow Object**? 👀
> 
> Yes, it is. In fact, I suggest you to add a new subtitle here clarifying with kind of flow the example is representing. I would add
> 
> ```
> 
> ###### clientCredentials Oauth Flow object example
> ```
> 
> Right after the `##### OAuth Flow Object Examples` at https://github.com/AnimeshKumar923/asyncapi-spec/blob/54fab4bd1f172a81e43ecd6a2058a4160f1f4e06/spec/asyncapi.md?plain=1#L2612

Got it. Will be adding this in [this PR](https://github.com/AnimeshKumar923/asyncapi-spec/pull/2) which is open against the `embedded-examples-validation` branch.

Reply:
> > > Maybe [this](https://github.com/AnimeshKumar923/asyncapi-spec/blob/54fab4bd1f172a81e43ecd6a2058a4160f1f4e06/spec/asyncapi.md?plain=1#L2614C1-L2634C52) is the potential fix to the **OAuth Flow Object**? 👀
> > 
> > 
> > Yes, it is. In fact, I suggest you to add a new subtitle here clarifying with kind of flow the example is representing. I would add
> > ```
> > 
> > ###### clientCredentials Oauth Flow object example
> > ```
> > Right after the `##### OAuth Flow Object Examples` at https://github.com/AnimeshKumar923/asyncapi-spec/blob/54fab4bd1f172a81e43ecd6a2058a4160f1f4e06/spec/asyncapi.md?plain=1#L2612
> 
> Got it. Will be adding this in [this PR](https://github.com/AnimeshKumar923/asyncapi-spec/pull/2) which is open against the `embedded-examples-validation` branch.

Implemented it [here](https://github.com/AnimeshKumar923/asyncapi-spec/blob/b0d36e7d35c023c8a67ed298e0320067fd5409c9/spec/asyncapi.md?plain=1#L2612C1-L2632C4)